client/src/modules/App.jsx

```diff
@@ -2,13 +2,56 @@ import React from 'react';
 import Nav from './Nav';
 import Home from './Home';
 
-export default React.createClass({
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loggedIn: '',
+    };
+
+    this.authenticateUser = this.authenticateUser.bind(this);
+  }
+
+  authenticateUser() {
+    var context = this;
+
+    fetch('http://127.0.0.1:3000/api/users/authenticate', {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+    })
+    .then(function(response) {
+      return response.json()
+    })
+    .then(function(jsonResponse) {
+      context.setState({loggedIn: jsonResponse.username});
+    })
+    .catch(function(err) {
+      console.log(err);
+    });
+  }
+
+  componentDidMount() {
+    !this.state.loggedIn ? this.authenticateUser() : null;
+  }
+
   render() {
     return (
       <div>
-        <Nav />
-        {this.props.children || <Home />}
+        <Nav loggedIn={this.state.loggedIn} auth={this.authenticateUser} />
+        {
+          React.cloneElement(
```

Comment:
Yup. There are a couple of ways I've seen but this seems to be the most elegant and least verbose. 

Ref: http://stackoverflow.com/questions/31862839/passing-props-to-react-router-children-routes